src/registry.rs

```diff
@@ -20,33 +21,93 @@ pub trait RegistryObserver {
     );
 }
 
+#[derive(Debug, Clone)]
+pub struct Config {
+    entry_ttl: Duration,
+}
+
+impl Config {
+    pub const ENTRY_TTL_MIN: Duration = Duration::from_secs(2);
+
+    pub fn entry_ttl(&self) -> Duration {
+        self.entry_ttl
+    }
+
+    pub fn set_entry_ttl_unchecked(self, value: Duration) -> Self {
```

Comment:
Why is this called unchecked? It's usually used to indicate that the code does not do validation of the input. For example: https://doc.rust-lang.org/stable/std/primitive.tuple.html#method.get_unchecked

Here we are validating the value against the Max.

Reply:
We are not really validating, which would mean returning a `Result<T, E>` if the value is invalid. We are simply overriding the value from the caller if it's unacceptable and so the caller cannot assume that the method will obey setting the input value exactly as provided. It is _range-bound_. If _unchecked_ doesn't sound right, I could suggest `set_entry_ttl_bounded` as an alternative. I wanted to leave room open for a checked version that's simply called `set_entry_ttl` and returns `Result<Self, InvalidDuration>`.

Reply:
Sounds great `set_entry_ttl_bounded` is clearer to me.